BigQuery/src/Model.php

```diff
@@ -0,0 +1,229 @@
+        $this->identity = [
+            'modelId' => $id,
+            'datasetId' => $datasetId,
+            'projectId' => $projectId
+        ];
+
+        $this->info = $info;
+    }
+
+    /**
+     * Retrieve's the model's details. If no model data is cached, a network request
+     * will be made to retrieve it.
+     *
+     * Example:
+     * ```
+     * $info = $model->info();
+     * echo $info['modelType'];
+     * ```
+     *
+     * @see https://cloud.google.com/bigquery/docs/reference/rest/v2/models#resource Model resource documentation.
+     * @param array $options [optional] Configuration options.
+     * @return array
+     */
+    public function info(array $options = [])
+    {
+        return $this->info ?: $this->reload($options);
+    }
+
+    /**
+     * Triggers a network request to reload the model's details.
+     *
+     * Example:
+     * ```
+     * $model->reload();
+     * $info = $model->info();
+     * echo $info['modelType'];
+     * ```
+     *
+     * @see https://cloud.google.com/bigquery/docs/reference/rest/v2/models/get Models get API documentation.
+     *
+     * @param array $options [optional] Configuration options.
+     * @return array
+     */
+    public function reload(array $options = [])
+    {
+        return $this->info = $this->connection->getModel($options + $this->identity);
+    }
+
+    /**
+     * Retrieves the model's ID.
+     *
+     * Example:
+     * ```
+     *  echo $model->id();
+     * ```
+     *
+     * @return string
+     */
+    public function id()
+    {
+        return $this->identity['modelId'];
+    }
+
+    /**
+     * Retrieves the model's identity.
+     *
+     * An identity provides a description of a resource that is nested in nature.
+     *
+     * Example:
+     * ```
+     * echo $model->identity()['modelId'];
+     * ```
+     *
+     * @return array
+     */
+    public function identity()
+    {
+        return $this->identity;
+    }
+
+    /**
+     * Delete the model.
+     *
+     * Please note that by default the library will not attempt to retry this
+     * call on your behalf.
+     *
+     * Example:
+     * ```
+     * $model->delete();
+     * ```
+     *
+     * @see https://cloud.google.com/bigquery/docs/reference/rest/v2/models/delete Models delete API documentation.
+     *
+     * @param array $options [optional] Configuration options.
+     */
+    public function delete(array $options = [])
+    {
+        $this->connection->deleteModel(
+            $options
+            + ['retries' => 0]
+            + $this->identity
+        );
+    }
+
+    /**
+     * Check whether or not the model exists.
+     *
+     * Example:
+     * ```
+     * echo $model->exists();
+     * ```
+     *
+     * @param array $options [optional] Configuration options.
+     * @return bool
+     */
+    public function exists(array $options = [])
+    {
+        try {
+            $this->connection->getModel($this->identity + $options);
```

Comment:
I know there are examples of direct connection calls in `exists()`, but I prefer calling `Model::reload($options)` here, and think we should make an effort to replace those eventually. Segregate connection calls as much as possible, and hydrate the instance in case this call is coming ahead of `info()`.

Reply:
I like that better. I can change the other parts of BigQuery in a future PR.